### DISCUSSION TITLE
Use as library

### DISCUSSION THREAD

**Discussion by sundarv85** [corazawaf/coraza#1016] [Q&A]
Hello,

I want to integrate coraza to only detect the issues on a fixed set of API Requests that we already log into a file. So I'm writing it as a go binary, which gets each of these API requests indvidiaully and then I want to run the OWASP rules against each request individually.

Can coraza be used as a library within my golang application? Any examples on how to achieve that.

**Comment by jcchavezs** [comment]
Totally, Coraza is a Go library you can use to build whatever you want :)

José Carlos Chávez


tor. 14. mars 2024 kl. 10:24 skrev Sundar ***@***.***>:

**Reply by sundarv85** [reply]
Thanks. Do you know of other applications that use Coraza as a library. That could give me some examples on how to use it.

**Comment by jptosso** [comment]
Yes, look at the examples dir, you will notice you have access to everything required to write your own implementation

**Reply by sundarv85** [reply]
@jptosso  Do you mean this one 

https://github.com/corazawaf/coraza/tree/main/examples/http-server

**Reply by jptosso** [reply]
Yes! You can use the http wrapper or write the transaction yourself

**Reply by sundarv85** [reply]
I’m trying to get information on

1. how to set my own transaction - as I have the URL, Query Params, Request Body
2. How to call the coraza with this manually set transaction and get the matching rules

https://pkg.go.dev/github.com/corazawaf/coraza/v3

this one doesn’t show about creating own transaction directly. 

https://pkg.go.dev/github.com/corazawaf/coraza

The above godoc shows an old 2021 package. Still it is not clear on how to set own transaction from here.

**Comment by jcchavezs** [comment]
Check https://pkg.go.dev/github.com/corazawaf/coraza/v3

In the official http middleware you can see how to populate the
request/response elements.

This is how you create a transaction:
https://github.com/corazawaf/coraza/blob/9184eeebf038decac73bf81616374c0932eb2ba7/http/middleware.go#L122
These are the methods you can use to populate the transaction with http
request/response components:
https://github.com/corazawaf/coraza/blob/9184eeebf038decac73bf81616374c0932eb2ba7/types/transaction.go#L19



José Carlos Chávez


tor. 14. mars 2024 kl. 14:23 skrev Sundar ***@***.***>:

**Reply by sundarv85** [reply]
Thanks a lot. I will try it and will reach out if I have any questions.
